examples-gallery/plot_sliding_block.py

```diff
@@ -0,0 +1,333 @@
+# %%
+""""
+Sliding a Block on a Road
+=========================
+A block, modeled as a particle is sliding on a road to cross a
+hill. The block is subject to gravity and speed dependent
+friction.
+Gravity points in the negative Y direction.
+A force tangential to the road is applied to the block.
+Three objective functions to me minimized may be selected:
+
+- selektion = 0: time to reach the end point is minimized
+- selektion = 1: time to reach the end point and the energy are minimized.
+- selektion = 2: energy consumed is minimized.
+
+**Constants**
+
+- m: mass of the block [kg]
+- g: acceleration due to gravity [m/s**2]
+- reibung: coefficient of friction [N/(m*s)]
+- a, b: paramenters determining the shape of the road.
+
+**States**
+
+- x: position of the block [m]
+- ux: velocity of the block [m/s]
+
+**Specifieds**
+
+- F: force applied to the block [N]
+
+"""
+import sympy.physics.mechanics as me
+from collections import OrderedDict
+import numpy as np
+import sympy as sm
+from opty.direct_collocation import Problem
+import matplotlib.pyplot as plt
+import matplotlib as mp
+from matplotlib.animation import FuncAnimation
+from copy import deepcopy
+# %%
+# define the road
```

Comment:
Use complete sentences with capitalization and punctuation.

Reply:
You mean something like: _The function below defines the road on which the block slides_ ?